client/my-sites/email/email-providers-stacked-comparison/provider-cards/google-workspace-card.tsx

```diff
@@ -110,67 +104,7 @@ const GoogleWorkspaceCard = ( {
 		)
 	);
 
-	const productIsDiscounted = hasDiscount( gSuiteProduct );
-
-	const priceWithInterval = (
-		<PriceWithInterval
-			cost={ gSuiteProduct?.cost ?? 0 }
-			currencyCode={ currencyCode ?? '' }
-			hasDiscount={ productIsDiscounted }
-			intervalLength={ intervalLength }
-			sale={ gSuiteProduct?.sale_cost ?? null }
-		/>
-	);
-
-	const standardPriceForIntervalLength = formatPrice(
-		gSuiteProduct?.cost ?? 0,
-		currencyCode ?? ''
-	);
-	const salePriceForIntervalLength = formatPrice(
-		gSuiteProduct?.sale_cost ?? 0,
-		currencyCode ?? ''
-	);
-
-	const discount = productIsDiscounted ? (
-		<div className="google-workspace-card__discount-with-renewal">
-			{ translate(
-				'%(discount)d%% off{{span}}, %(discountedPrice)s billed today, renews at %(standardPrice)s{{/span}}',
-				{
-					args: {
-						discount: gSuiteProduct?.sale_coupon?.discount,
-						discountedPrice: salePriceForIntervalLength,
-						standardPrice: standardPriceForIntervalLength,
-					},
-					comment:
-						"%(discount)d is a numeric percentage discount (e.g. '50'), " +
-						"%(discountedPrice)s is a formatted, discounted price that the user will pay today (e.g. '$3'), " +
-						"%(standardPrice)s is a formatted price (e.g. '$5')",
-					components: {
-						span: <span />,
-					},
-				}
-			) }
-
-			<InfoPopover position="right" showOnHover>
-				{ translate(
-					'This discount is only available the first time you purchase a %(googleMailService)s account, any additional mailboxes purchased after that will be at the regular price.',
-					{
-						args: {
-							googleMailService: googleWorkspace.productName,
-						},
-						comment: '%(googleMailService)s can be either "G Suite" or "Google Workspace"',
-					}
-				) }
-			</InfoPopover>
-		</div>
-	) : null;
-
-	googleWorkspace.priceBadge = (
-		<PriceBadge
-			additionalPriceInformationComponent={ discount }
-			priceComponent={ priceWithInterval }
-		/>
-	);
+	googleWorkspace.priceBadge = <GoogleWorkspacePrice intervalLength={ intervalLength } />;
```

Comment:
I like a lot this movement, but I am wondering if it should be a more generic component (AKA `PriceBadge`) and be used with a few input parameters so it can be reusable for both providers.

I would say that keeping it as singular component for each provider is better, as it would need a lot of code to tune it to make it generic. I'm just dropping my thoughts 🙂

Reply:
I went that way because the benefits of having one component for both email providers wasn't clear to me. I actually think this would increase the number of input parameters.